### DISCUSSION TITLE
Made some of the build process multithreaded

### DISCUSSION THREAD

**Discussion by SwedishBarbarossa** [valhalla/valhalla#4841] [Show and tell]
https://github.com/valhalla/valhalla/commit/0f2936f0fa349fd81ea8824fa8bbafeeadb60084

It compiles (for me :joy:), and processing seems to be in order, but I couldn't quite figure out how unit tests work.

I wouldn't call myself a c++ programmer and I kinda need to move on from this in a half finished state.

What I think needs to be done from here:
1. Make sure it actually works and doesn't corrupt the data or crash repeatedly etc.
2. Make sure it doesn't leak memory.
3. Fetch the configured thread count where the Parse function is called. I hard-coded them as 4 in this commit. I'm not 100% sure I got all of them - but the compiler should have complained if I missed one.
4. I think I forgot to remove unused imports :cold_sweat:.
5. Get it to adhere to the repo style.

Even if its just a jumping off place to see what actually compiled and worked before you implement a better solution, I thought it was a waste not to share before I move on.

edit: kinda started based on this exchange https://github.com/valhalla/valhalla/issues/4801

**Comment by kevinkreiser** [comment]
neat thanks for sharing, getting it as a pr even if we cant merge it might be nice to keep it on our minds over time.

**Reply by SwedishBarbarossa** [reply]
[Here](https://github.com/valhalla/valhalla/pull/4842) :saluting_face: 

Godspeed however you end up using it. Awesomely useful project for what I'm working on currently.
